RFS/src/main/java/com/rfs/worker/DocumentsRunner.java

```diff
@@ -18,10 +18,10 @@ public class DocumentsRunner implements Runner {
 
     private final DocumentsStep.SharedMembers members;
 
-    public DocumentsRunner(GlobalState globalState, CmsClient cmsClient, String snapshotName, IndexMetadata.Factory metadataFactory,
-                ShardMetadata.Factory shardMetadataFactory, SnapshotShardUnpacker unpacker, LuceneDocumentsReader reader,
-                DocumentReindexer reindexer) {
-        this.members = new DocumentsStep.SharedMembers(globalState, cmsClient, snapshotName, metadataFactory, shardMetadataFactory, unpacker, reader, reindexer);
+    public DocumentsRunner(GlobalState globalState, CmsClient cmsClient, String snapshotName, long maxShardSizeBytes,
```

Comment:
This is nigh impossible to read - lets move to using @requireargsconstructor as a quick stop gap, but I think we need to have better data modeling so we don't have to pass every configuration option around, clustering by functionality or having a settings object that can be peaked into would handle this

Reply:
I don't find it hard to read, but feel free to update it how you'd like.

Reply:
In general, I go back and forth on this one in other areas of Java code.  Doing the 'right' thing with builders is cumbersome and it only amounts to a runtime check.
Please try to put these on separate lines and order them to tell the best story possible.  This is basically dependency injection and everything is required, right?

Reply:
@gregschohn Yeah, this is just dependency injection.  I can put them on new lines if that makes it a bit clearer, but the bigger issue is that there's something that feels off with how the SharedMembers object is created in an intermediate scope between the main() and the Step classes that need it.  It feels like the SharedMembers definition might not actually belong in the Step class?